frontend/lib/resolution/InitResolver.cpp

```diff
@@ -607,6 +623,27 @@ bool InitResolver::handleCallToSuperInit(const FnCall* node,
   return false;
 }
 
+void InitResolver::updateSuperType(const CallResolutionResult* c) {
+  auto& msc = c->mostSpecific().only();
+  auto superThis = msc.formalActualMap().byFormalIdx(0).formalType();
+
+  const BasicClassType* superType = nullptr;
+  if (auto ct = superThis.type()->toClassType()) {
```

Comment:
Nitpick but I think it's cleaner to replace this `if` with `superType = superThis->getCompositeType()->toBasicClassType()`